src/library/playlisttablemodel.cpp

```diff
@@ -170,7 +175,16 @@ void PlaylistTableModel::setTableModel(int playlistId) {
             LIBRARYTABLE_ID,
             columns,
             m_pTrackCollectionManager->internalCollection()->getTrackSource());
-    setSearch("");
+
+    // Restore search text
+    QString prevSearch("");
+    if (m_iPlaylistId != -1) {
+        auto it = m_searchTexts.constFind(m_iPlaylistId);
+        if (it != m_searchTexts.constEnd()) {
+            prevSearch = it.value();
+        }
+    }
+    setSearch(prevSearch);
```

Comment:
I think we can simplify this even more, because m_iPlaylistId will never be -1 and even if, the QHash will return an empty QString: 
```suggestion
    setSearch(m_searchTexts.value(m_iPlaylistId));
```

Reply:
I tried that first but it's not working because in the called code the string is tested for `isNull` (not `isEmty`).
I can check whether `isEmpty` can be used down the line.

Reply:
QString() didn't work because of
https://github.com/mixxxdj/mixxx/blob/3aa2318e34f306808420a6707857dabd37d0bea2/src/widget/wsearchlineedit.cpp#L456-L464

oh, this was you, eight years ago : ) 
https://github.com/mixxxdj/mixxx/commit/54f71ea7b8e5d6e87ad9132bcf7fdfe2a7c8372a#diff-5112567721f7d2345d3df3bdf1e73badf29f2feb7796f069777b54b949c910feL120

I'll remove the isNull branch. It is not required anymore since `disableSearch` is called explicitly where required.

Reply:
done.